crates/bitwarden-ssh/src/generator.rs

```diff
@@ -0,0 +1,87 @@
+use serde::{Deserialize, Serialize};
+use ssh_key::{rand_core::CryptoRngCore, Algorithm};
+#[cfg(feature = "wasm")]
+use tsify_next::Tsify;
+
+use crate::{error, error::KeyGenerationError, SshKey};
+
+#[derive(Serialize, Deserialize)]
+#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+pub enum KeyAlgorithm {
+    Ed25519,
+    Rsa3072,
+    Rsa4096,
+}
+
+/**
+ * Generate a new SSH key pair, for the provided key algorithm, returning
+ * an `SshKey` struct containing the private key, public key, and key fingerprint,
```

Comment:
**question:** Link?
```suggestion
 * an [SshKey] struct containing the private key, public key, and key fingerprint,
```

Reply:
Added, thogh clicking them just brings me to a not found page: https://docs.rs/bitwarden_ssh/1.0.0/bitwarden_ssh/struct.SshKey.html

Reply:
It should work when you generate docs locally or use rust analyzer in vscode.